modules/vault-cluster/main.tf

```diff
@@ -11,42 +11,63 @@ terraform {
 # ---------------------------------------------------------------------------------------------------------------------
 
 resource "aws_autoscaling_group" "autoscaling_group" {
-  name_prefix = "${var.cluster_name}"
+  name_prefix = var.cluster_name
 
-  launch_configuration = "${aws_launch_configuration.launch_configuration.name}"
+  launch_configuration = aws_launch_configuration.launch_configuration.name
 
-  availability_zones  = ["${var.availability_zones}"]
-  vpc_zone_identifier = ["${var.subnet_ids}"]
+  availability_zones  = var.availability_zones
+  vpc_zone_identifier = var.subnet_ids
 
   # Use a fixed-size cluster
-  min_size             = "${var.cluster_size}"
-  max_size             = "${var.cluster_size}"
-  desired_capacity     = "${var.cluster_size}"
-  termination_policies = ["${var.termination_policies}"]
+  min_size             = var.cluster_size
+  max_size             = var.cluster_size
+  desired_capacity     = var.cluster_size
+  termination_policies = [var.termination_policies]
 
-  health_check_type         = "${var.health_check_type}"
-  health_check_grace_period = "${var.health_check_grace_period}"
-  wait_for_capacity_timeout = "${var.wait_for_capacity_timeout}"
+  health_check_type         = var.health_check_type
+  health_check_grace_period = var.health_check_grace_period
+  wait_for_capacity_timeout = var.wait_for_capacity_timeout
 
-  enabled_metrics = ["${var.enabled_metrics}"]
+  enabled_metrics = var.enabled_metrics
 
   # Use bucket and policies names in tags for depending on them when they are there
   # And only create the cluster after S3 bucket and policies exist
   # Otherwise Vault might boot and not find the bucket or not yet have the necessary permissions
   # Not using `depends_on` because these resources might not exist
-  tags = ["${concat(
-    list(
-      map(
-        "key", var.cluster_tag_key,
-        "value", var.cluster_name,
-        "propagate_at_launch", true,
-        "using_s3_bucket_backend",  element(concat(aws_iam_role_policy.vault_s3.*.name, list("")), 0),
-        "s3_bucket_id", element(concat(aws_s3_bucket.vault_storage.*.id, list("")), 0),
-        "using_auto_unseal",  element(concat(aws_iam_role_policy.vault_auto_unseal_kms.*.name, list("")), 0),
-      )
-    ),
-    var.cluster_extra_tags)
-  }"]
+  tag {
+    key                 = var.cluster_tag_key
+    value               = var.cluster_name
+    propagate_at_launch = true
+  }
+
+  tag {
+    key                 = "using_s3_bucket_backend"
+    value               = element(concat(aws_iam_role_policy.vault_s3.*.name, list("")), 0)
```

Comment:
Shouldn't it be `[""]` instead of `list("")`?

Reply:
Ah, yes, this is indeed deprecated in 0.12.x 

https://www.terraform.io/docs/configuration/functions/list.html

Reply:
88aa91789c20222b84927551844f0176b69a312f